jenkins/data-prepper/promote-distribution-artifacts.jenkinsfile

```diff
@@ -0,0 +1,53 @@
+lib = library(identifier: 'jenkins@20211123', retriever: legacySCM(scm))
+
+pipeline {
+    agent {
+        docker {
+            label 'Jenkins-Agent-al2-x64-c54xlarge-Docker-Host'
+            image 'opensearchstaging/ci-runner:ci-runner-centos7-v1'
+            alwaysPull true
+        }
+    }
+    parameters {
+        string(name: 'VERSION', description: 'The version of Data Prepper', trim: true)
+        string(name: 'SOURCE_JOB_BASE_NAME', description: 'The name of the artifacts job', trim: true)
+        string(name: 'SOURCE_BUILD_NUMBER', description: 'The build number of the artifacts job', trim: true)
+    }
+    stages {
+        stage('Download Staging Artifacts') {
+            steps {
+                script {
+                    downloadPath = ([
```

Comment:
Since the time between the first job being run and this job being run should typically be small, using artifacts might be a cleaner mechanism

- Save artifacts, https://www.jenkins.io/doc/pipeline/steps/workflow-basic-steps/#archive-archive-artifacts
- Unarchive https://www.jenkins.io/doc/pipeline/steps/workflow-basic-steps/#unarchive-copy-archived-artifacts-into-the-workspace
- Examples https://medium.com/@gustavo.guss/jenkins-archive-artifact-save-file-in-pipeline-ac6d8b569c2c

The concern I have is that the construction of those paths can change, things can be added and it isn't apparent that this job needs to be updated.  If the input to this function was the build to

Reply:
Yes, I share you concerns on this.

Because Data Prepper needs a manual smoke test step (presently), I took the approach of uploading the staging artifacts to S3. This way, somebody on our team can easily download them and test them. I'm concerned about having difficulty getting access to Jenkins artifacts instead of S3 objects. (Unless we use something like this plugin: https://plugins.jenkins.io/s3/)

If I take the approach you suggested with the `input` parameter, we would have both: 1) The files available for download from an S3 bucket; and 2) The path-generation code isolated one job.